cli/internal/turbopath/anchored_system_path.go

```diff
@@ -44,3 +47,22 @@ func (p AnchoredSystemPath) Join(additional ...RelativeSystemPath) AnchoredSyste
 	cast := RelativeSystemPathArray(additional)
 	return AnchoredSystemPath(filepath.Join(p.ToString(), filepath.Join(cast.ToStringArray()...)))
 }
+
+// HasPrefix is strings.HasPrefix for paths, ensuring that it matches on separator boundaries.
+// This does NOT perform Clean in advance.
+func (p AnchoredSystemPath) HasPrefix(prefix AnchoredSystemPath) bool {
+	prefixLen := len(prefix)
+	pathLen := len(p)
+
+	if prefixLen > pathLen {
+		// Can't be a prefix if longer.
+		return false
+	} else if prefixLen == pathLen {
+		// Can be a prefix if they're equal, but otherwise no.
+		return p == prefix
+	}
+
+	// otherPath is definitely shorter than p.
+	// We need to confirm that p[len(otherPath)] is a system separator.
+	return os.IsPathSeparator(p[prefixLen])
```

Comment:
Won't this return true for `"foo/bar".HasPrefix("baz/")` or for any prefix that has a length that happens to line up with a path separator?

Reply:
Yup. Totally missed a step here.